src/Repository/SessionRepository.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace App\Repository;
+
+use App\Entity\Session;
+use Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepository;
+use Symfony\Bridge\Doctrine\RegistryInterface;
+
+/**
+ * @method Session|null find($id, $lockMode = null, $lockVersion = null)
+ * @method Session|null findOneBy(array $criteria, array $orderBy = null)
+ * @method Session[]    findAll()
+ * @method Session[]    findBy(array $criteria, array $orderBy = null, $limit = null, $offset = null)
+ */
+class SessionRepository extends ServiceEntityRepository
+{
+    public function __construct(RegistryInterface $registry)
+    {
+        parent::__construct($registry, Session::class);
+    }
+
+    /**
+     * @param $day
+     * @return mixed
+     */
+    public function findByDayField($day)
+    {
+        return $this->createQueryBuilder('s')
+            ->select('s.startsAt')->distinct()
+            ->andWhere('s.day = :val')
+            ->setParameter('val', $day)
+            ->getQuery()
+            ->getResult()
+        ;
+    }
+
+
+    /**
+     * @param $day
+     * @return mixed
+     */
+    public function findFreeTimeSlotByDate($day)
+    {
+        return $this->createQueryBuilder('s')
+            ->andWhere('s.status = :val')
+            ->andWhere('s.reservedAt = :day')
+            ->setParameter('val', 'free')
+            ->setParameter('day', $day)
+            ->getQuery()
+            ->getResult()
+            ;
+    }
+
+
+    /**
+     * @return mixed
+     */
+    public function init_calendar()
+    {
+        return $this->createQueryBuilder('s')
+            ->andWhere('s.status = :val')
+            ->setParameter('val', 'free')
+            ->getQuery()
+            ->getResult()
+            ;
+    }
+
+    public function checkSlotFree($date, $startTime, $endTime)
+    {
+        return $this->createQueryBuilder('s')
+            ->select('s.id')
+            ->andWhere('s.status = :val')
```

Comment:
kaip ir rasiau virsuj. Realiai - visi šitie metodai - niekiniai kaip ir pati klasė, tiek kad SF4 rekomenduoja tau turėti custom'inę repo klasė kiekvienam entičiui, bet realybėje tai daugiau išimtis nei taisyklė, nes visi šitos repo metodai nereikalingi :)